### PR TITLE
Enable Logging Tick Events to Console with URL Parameter

### PR DESCRIPTION
**Pull request by thsparks**
Add a url parameter (`consoleticks`), which will send all tick events to the console. I find this makes it much easier to test and debug telemetry events.

It comes with a two "modes". By default (`consoleticks=1` or `consoleticks=verbose`), it will print output like so:
```15:34:29 - Tick - activity.autorun {"editor":"blocksArea"} <no opts>```

You can also specify `consoleticks=2` or `consoleticks=short` to get a more abbreviated output:
```activity.autorun {"editor":"blocksArea"} <no opts>```

### CODE REVIEW

File: pxtlib/analytics.ts [hunk 1]
```diff
@@ -33,6 +33,13 @@ namespace pxt.analytics {
 
         const te = pxt.tickEvent;
         pxt.tickEvent = function (id: string, data?: Map<string | number>, opts?: TelemetryEventOptions): void {
+            // Log tick to console if "consoleticks" url param is present.
+            // We do this simple check first (rather than the full parse) to minimize overhead when it's not present, which is most of the time.
+            if(window && /consoleticks=/.test(window.location.href))
```
Comment (jwunderl) [inline pxtlib/analytics.ts:38]:
```suggestion
            if (window && /consoleticks=/.test(window.location.href))
```

**Review by jwunderl — APPROVED** [review]
lgtm!

**Review by kimprice — APPROVED** [review]
Nice! This will be very useful. :)

File: pxtlib/analytics.ts [hunk 2]
```diff
@@ -33,6 +33,13 @@ namespace pxt.analytics {
 
         const te = pxt.tickEvent;
         pxt.tickEvent = function (id: string, data?: Map<string | number>, opts?: TelemetryEventOptions): void {
+            // Log tick to console if "consoleticks" url param is present.
+            // We do this simple check first (rather than the full parse) to minimize overhead when it's not present, which is most of the time.
+            if (window && /consoleticks=/.test(window.location.href))
```
Comment (eanders-ms) [inline pxtlib/analytics.ts:38]: I'd prefer this were controlled by a flag set by the application, e.g. `pxt.analytics.consoleTicks = true`. How the flag is set is up to the app.

Comment (jwunderl) [reply pxtlib/analytics.ts:38]: Here's where we have a few similar flags set to this:

https://github.com/microsoft/pxt/blob/stable8.5/webapp/src/app.tsx#L5539

Comment (thsparks) [reply pxtlib/analytics.ts:38]: @eanders-ms - just wanted to clarify, do you mean it should still be enabled/disabled via a query parameter, but that query parameter should then set a flag (like `pxt.analytics.consoleTicks`), or do you mean we should just have a flag hardcoded to false which devs can set to true if they want to debug? Or something else entirely?

Comment (eanders-ms) [reply pxtlib/analytics.ts:38]: pxtlib should be agnostic to how the flag is set. Leave that to the app. The webapp would parse the query param and set the flag. The cli would read a command line arg.

**Review by eanders-ms — APPROVED** [review]
Looks good!

**Review by jwunderl — APPROVED** [review]

**Pull request merged**